devito/passes/iet/engine.py

```diff
@@ -398,18 +398,21 @@ def abstract_efunc(efunc):
     return efunc
 
 
-def abstract_objects(objects, sregistry=None):
+def abstract_objects(objects0, sregistry=None):
     """
     Proxy for `abstract_object`.
     """
+    # Expose hidden objects for complete reconstruction
+    objects = []
+    for i in objects0:
+        if i.is_Bundle:
+            objects.extend(i.components)
+        objects.append(i)
+
     # Precedence rules make it possible to reconstruct objects that depend on
     # higher priority objects
-    priority = {
-        Array: 1,
-        DiscreteFunction: 2,
-        AbstractIncrDimension: 3,
-        BlockDimension: 4,
-    }
+    keys = [Bundle, Array, DiscreteFunction, AbstractIncrDimension, BlockDimension]
```

Comment:
Nitpick: worth making this a tuple?

Reply:
note that these comments belong to the parent PR as per this PR's description